### DISCUSSION TITLE
Skip n lines until you reach the CSV data

### DISCUSSION THREAD

**Discussion by jbertovic** [nushell/nushell#4146] [Q&A] [answered]
hello all,

I have a csv file where the header and data don't start until line 7.  I've tried a variety of methods of trying to skip the first 7 lines but can't get back to csv data. 

tried:
`open file.csv -r | lines | skip 7 | to csv `

but gives me all the data in one long string instead of broken out in table.  'open' without the -r flag can't recognize the file as csv.

Any ideas - or do i just pre-process the file using some other command like 'sed' or script?

**Comment by hustcer** [comment, marked as answer]
Try `open file.csv -r | lines | skip 7 | str collect $'(char nl)' | from csv -n`? If you have a header at the 8th line use: `open file.csv -r | lines | skip 7 | str collect $'(char nl)' | from csv`

**Reply by jbertovic** [reply]
I tried both `to csv` and `from csv` without luck.  `from csv` doesn't return anything. `to csv` will return all the data in one block that is quoted.

It looks like stopping at `skip 7` just gives me a table with one column that holds each lines csv data.  I'm running this in windows....I could try this on linux to see if it comes out differently.

**Reply by hustcer** [reply]
Try `open file.csv -r | lines | skip 7 | str collect $'(char nl)' | from csv -n`?
 If you have a header at the 8th line use: `open file.csv -r | lines | skip 7 | str collect $'(char nl)' | from csv`

**Reply by jbertovic** [reply]
muchly appreciated...need to read what that does as I don't understand taking a quick look at it.  Unless you have time to explain.  

Thanks again.
